sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/README.md

```diff
@@ -0,0 +1,100 @@
+# Azure Web PubSub service client subscriber library for .NET
+
+[Azure Web PubSub Service](https://aka.ms/awps/doc) is a service that enables you to build real-time messaging web applications using WebSockets and the publish-subscribe pattern. Any platform supporting WebSocket APIs can connect to the service easily, e.g. web pages, mobile applications, edge devices, etc. The service manages the WebSocket connections for you and allows up to 100K **concurrent** connections. It provides powerful APIs for you to manage these clients and deliver real-time messages.
+
+Any scenario that requires real-time publish-subscribe messaging between server and clients or among clients, can use Azure Web PubSub service. Traditional real-time features that often require polling from server or submitting HTTP requests, can also use Azure Web PubSub service.
+
+This library can be used to do the following actions. Details about the terms used here are described in [Key concepts](#key-concepts) section.
+
+- Parse upstream request under CloudNative CloudEvents
+- Add validation options for upstream request
+- API to add user defined functionality to handle different upstream events
+
+[Source code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src) |
+[Package](https://www.nuget.org/packages/Microsoft.Azure.WebPubSub.AspNetCore) |
+[API reference documentation](https://aka.ms/awps/sdk/csharp) |
+[Product documentation](https://aka.ms/awps/doc) |
+[Samples][samples_ref]
+
+## Getting started
+
+### Install the package
+
+Install the client library from [NuGet](https://www.nuget.org/):
+
+```PowerShell
+dotnet add package Microsoft.Azure.WebPubSub.AspNetCore --prerelease
+```
+
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+- An existing Azure Web PubSub service instance.
+
+### Authenticate the client
+
+In order to interact with the service, you'll need to create an instance of the WebPubSubServiceClient class. To make this possible, you'll need the connection string or a key, which you can access in the Azure portal.
+
+### Create a `WebPubSubServiceClient`
+
+```C# Snippet:WebPubSubAuthenticate
+var serviceClient = new WebPubSubServiceClient(new Uri(endpoint), "some_hub", new AzureKeyCredential(key));
+```
+
+## Key concepts
+
+For information about general Web PubSub concepts [Concepts in Azure Web PubSub](https://docs.microsoft.com/azure/azure-web-pubsub/key-concepts)
+
+## Examples
+
+### Add upstream validation options
+
+```C# Snippet:WebPubSubValidationOptions
+var wpsHandler = new WebPubSubRequestBuilder()
+    .AddValidationOptions(new WebPubSubValidationOptions("<connection-string>"))
```

Comment:
Is the intent here to validate that requests you are receiving are coming from the expected WPS instance?

Reply:
Yes. Endpoints will be used in `AbuseProtection` validation and `AccessKey` will be used to calculate Signature.